editor/gui/editor_scroll_box.cpp

```diff
@@ -0,0 +1,199 @@
+
+#include "editor_scroll_box.h"
+
+#include "scene/gui/button.h"
+#include "scene/gui/scroll_bar.h"
+#include "scene/gui/scroll_container.h"
+#include "scene/theme/theme_db.h"
+
+// Queue call this method to ensure the control is still valid before calling ScrollContainer::ensure_control_visible.
+// See https://github.com/godotengine/godot/issues/109496.
+void EditorScrollBox::ensure_control_visible(Control *p_control) {
+	if (!p_control || !scroll_container->is_ancestor_of(p_control)) {
+		return;
+	}
+
+	scroll_container->ensure_control_visible(p_control);
+}
+
+// Use set_control(nullptr) to remove the control used by the ScrollContainer.
+void EditorScrollBox::set_control(Control *p_control) {
+	if (p_control) {
+		ERR_FAIL_COND_MSG(has_control(), "Container already has a control, use set_control(nullptr) to remove it.");
+	}
+	if (control) {
+		control->disconnect(SceneStringName(resized), callable_mp(this, &EditorScrollBox::_update_buttons));
+		scroll_container->remove_child(p_control);
+		control = nullptr;
+	}
+
+	if (!p_control) {
+		_update_buttons();
+		return;
+	}
+
+	if (p_control->get_parent()) {
+		p_control->get_parent()->remove_child(p_control);
+	}
+
+	control = p_control;
+	control->connect(SceneStringName(resized), callable_mp(this, &EditorScrollBox::_update_buttons));
+	scroll_container->add_child(p_control);
+
+	_update_buttons();
+}
+
+bool EditorScrollBox::has_control() const {
+	return control != nullptr;
+}
+
+Button *EditorScrollBox::get_right_button() const {
+	return is_layout_rtl() ? left_button : right_button;
+}
+
+Button *EditorScrollBox::get_left_button() const {
+	return is_layout_rtl() ? right_button : left_button;
+}
+
+void EditorScrollBox::_scroll(bool p_right) {
+	if (is_layout_rtl()) {
+		p_right = !p_right;
+	}
+
+	HScrollBar *hscroll_bar = scroll_container->get_h_scroll_bar();
+	if (Input::get_singleton()->is_key_pressed(Key::CTRL)) {
+		hscroll_bar->set_value(p_right ? hscroll_bar->get_max() : 0);
+	} else if (Input::get_singleton()->is_key_pressed(Key::SHIFT)) {
+		hscroll_bar->set_value(hscroll_bar->get_value() + hscroll_bar->get_page() * (p_right ? 1 : -1));
+	} else {
+		hscroll_bar->set_value(hscroll_bar->get_value() + (hscroll_bar->get_page() * 0.5) * (p_right ? 1 : -1));
+	}
+}
+
+// Update the buttons visibility and disabled state.
+void EditorScrollBox::_update_buttons() {
+	bool show_arrows = control && control->get_size().width > scroll_container->get_size().width;
+	left_button->set_visible(show_arrows);
+	right_button->set_visible(show_arrows);
+
+	if (show_arrows) {
+		_update_disabled_buttons();
+	}
+}
+
+void EditorScrollBox::_update_buttons_icon_and_tooltip() {
+	Button *button = get_left_button();
+	button->set_button_icon(theme_cache.arrow_left);
+	button->set_accessibility_name(TTRC("Scroll Left"));
+	button->set_tooltip_text(TTRC("Scroll Left\nHold Ctrl to scroll to the begin.\nHold Shift to scroll one page."));
```

Comment:
On RTL layout it says scroll to the `begin` instead of `end`, and to fix this it will result in 4 translated lines instead of just 2, so we can consider replacing both translations to say `Hold Ctrl to scroll to the far left` and `Hold Ctrl to scroll to the far right`.